File: samples/AdventureWorksLT/UdtNameAttribute.cs
using System;
using DevZest.Data;
using DevZest.Data.MySql;
using DevZest.Data.Annotations.Primitives;
using DevZest.Data.Addons;

namespace DevZest.Samples.AdventureWorksLT
{
    [ModelDesignerSpec(addonTypes: new Type[] { typeof(ColumnNotNull), typeof(MySqlType) }, validOnTypes: new Type[] { typeof(_String) })]
    public sealed class UdtNameAttribute : ColumnAttribute
    {
        protected override void Wireup(Column column)
        {
            if (column is _String stringColumn)
            {
                stringColumn.Nullable(true);
                stringColumn.AsMySqlVarChar(50);
            }
        }
    }
}

File: samples/AdventuresWorksLT.VisualBasic/AdventureWorksLT/UdtNameAttribute.vb
Imports DevZest.Data.Annotations.Primitives
Imports DevZest.Data.Addons

<ModelDesignerSpec(New Type() {GetType(ColumnNotNull), GetType(SqlType)}, New Type() {GetType(_String)})>
Public NotInheritable Class UdtNameAttribute
    Inherits ColumnAttribute

    Protected Overrides Sub Wireup(ByVal column As Column)
        If TypeOf column Is _String Then
            column.Nullable(True)
            CType(column, _String).AsSqlNVarChar(50)
        End If
    End Sub
End Class
